cmd/ko/README.md

```diff
@@ -190,6 +186,31 @@ to whatever `kubectl` context is active.
 of convenience for cleaning up resources created through `ko apply`.
 
 
+## With DockerHub
+
+Unfortunately, DockerHub does not support this sort of multi-level names, so you may see an error like:
+
+```shell
+$ KO_DOCKER_REPO=docker.io/mattmoor ko publish ./cmd/crane
+2018/07/19 03:25:56 Using base gcr.io/distroless/base:latest for github.com/google/go-containerregistry/cmd/crane
+2018/07/19 03:25:56 Publishing index.docker.io/mattmoor/github.com/google/go-containerregistry/cmd/crane:latest
+2018/07/19 03:25:57 error publishing github.com/google/go-containerregistry/cmd/crane: UNAUTHORIZED: "authentication required"
+```
+
+To support this, we have a flag that will flatten the import path to `{package}-{hash of import path}`:
+
+```shell
+$ KO_DOCKER_REPO=docker.io/mattmoor ko publish --flat ./cmd/crane
```

Comment:
`--flat` is not very intuitive for users even though technically telling.

Also if you set `KO_DOCKER_REPO` couldn't the `--flat` be implicit ?